.github/workflows/ci.yml

```diff
@@ -0,0 +1,23 @@
+name: Verify
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 8, 11, 17 ]
+    steps:
+      - uses: actions/checkout@v3
+      - name: "Set up JDK ${{ matrix.java }}"
+        uses: actions/setup-java@v3
+        with:
+          java-version: ${{ matrix.java }}
+          distribution: 'temurin'
+      - name: Build & Test
+        run: ./mvnw -B verify -P run-its
```

Comment:
this repo does not have a lot of tests but it probably doesn't hurt to append a test summary action:

```yml
      - name: Create Test Summary
        uses: mikepenz/action-junit-report@v4
        if: always()
        with:
          annotate_only: true
          detailed_summary: true
          report_paths: 'nb*/target/surefire-reports/TEST-*.xml'
```
(I hope the path works - not tested)

we use  `test-summary/action@v2` in the main repo which has slightly different presentation - both have pros and cons (https://github.com/apache/netbeans/pull/6462). We could try `action-junit-report` here unless someone has a favorite.

Reply:
I think, as I have no experience with this action, to leave it for subsequent PR... make it work first, then beautify it next :smile: